testsuite/shared/src/main/java/org/jboss/as/test/deployment/DeploymentInfoUtils.java

```diff
@@ -0,0 +1,797 @@
+
+        // Error status
+        UNKNOWN("!--unknown--!"); // Default value if isn't found
+
+        private String title;
+
+        DeploymentState(String title) {
+            this.title = title;
+        }
+
+        public String getTitle() {
+            return title;
+        }
+
+        /**
+         * Return information about enabled or disabled state from deployment info command statuses
+         *
+         * @param status Status of application deployment
+         * @return true application deployment id enabled, false is disabled
+         */
+        public static boolean mapBooleanByDeploymentStatus(@NotNull DeploymentState status) {
+            switch (status) {
+                case OK:
+                case ENABLED:
+                    return true;
+                case ADDED:
+                case STOPPED:
+                    return false;
+                case NOT_ADDED:
+                case UNKNOWN:
+                default:
+                    throw new IllegalArgumentException("Unsupported state " + status + "!");
+            }
+        }
+    }
+
+    // #### BEGIN Internal containers classes
+
+    /**
+     * Represent container of called command deployment list/info.
+     * Parsing command output for processing.
+     * Hold additional information about processing verify check.
+     */
+    public static class DeploymentInfoResult {
+        // Called command list or info about applications deployments for checking
+        private final String command;
+        // Selected server group in list/info command
+        private final String serverGroup;
+        // Original output of list/info command
+        private final String originalOutput;
+        /* Holding output of called command for multiple checking output without recall command
+           Output is parsed for processing*/
+        private final List<String> rows;
+        // Represent request of management trusted command to verify check
+        private String request;
+        // Represent response of management trusted command to verify check
+        private String response;
+
+        private DeploymentInfoResult(String command, String serverGroup, String output) {
+            this.command = command;
+            this.serverGroup = serverGroup;
+            this.originalOutput = output;
+            if (output != null && !output.isEmpty()) {
+                log.info("Read output:\n" + output);
+                this.rows = Arrays.asList(output.split("\n"));
+            } else {
+                log.info("Read output: " + OUTPUT_EMPTY_MARK);
+                this.rows = new ArrayList<>();
+                this.rows.add(OUTPUT_EMPTY_MARK);
+            }
+        }
+
+        /**
+         * Checking if called command has empty output.
+         *
+         * @return If command output is empty return true, else If command has some output return false
+         */
+        public boolean isOutputEmpty() {
```

Comment:
a method isXXX is not expected to throw an exception. You should name it: checkOutputNotEmpty().
Ideally, if you expect DeploymentInfoResult instances to never reference empty output, you should make this check private and called in constructor.

Reply:
Is method isXXX must be safe? Ok.
Thank you. Done